File: fixtures/matlab/put_xput.m
%% Web Access using Data Import and Export API
uri = 'http://localhost:9200/twitter/_mapping/user?pretty';
body = struct(...
    'properties', struct(...
        'email', struct(...
            'type', 'keyword'...
        )...
    )...
);
options = weboptions(...
    'RequestMethod', 'put',...
    'MediaType', 'application/json'...
);
response = webwrite(uri, body, options);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

header = HeaderField('Content-Type', 'application/json');
uri = URI('http://localhost:9200/twitter/_mapping/user?pretty');
body = JSONProvider(struct(...
    'properties', struct(...
        'email', struct(...
            'type', 'keyword'...
        )...
    )...
));
response = RequestMessage('put', header, body).send(uri.EncodedURI);

File: fixtures/elixir/put_xput.ex
request = %HTTPoison.Request{
  method: :put,
  url: "http://localhost:9200/twitter/_mapping/user?pretty",
  options: [],
  headers: [
    {~s|Content-Type|, ~s|application/json|},
  ],
  params: [],
  body: ~s|{"properties": {"email": {"type": "keyword"}}}|
}

response = HTTPoison.request(request)
